### ISSUE TITLE
Node crash on listtransactions

### ISSUE THREAD

**Issue by AutonomousOrganization** [ElementsProject/lightning#6185]
Node crashes when calling `listtransactions` on latest version v23.02.2 - caused by some strangeness in my nodes history. Can give more information about the current state of my nodes db if you let me know what could help. Attached error:

```
malloc(): invalid next size (unsorted)
lightningd: FATAL SIGNAL 6 (version v23.02.2)
0x562720805e1f send_backtrace
	common/daemon.c:33
0x562720805eb6 crashdump
	common/daemon.c:46
0x7f5b0324251f ???
	./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7f5b03296a7c __pthread_kill_implementation
	./nptl/pthread_kill.c:44
0x7f5b03296a7c __pthread_kill_internal
	./nptl/pthread_kill.c:78
0x7f5b03296a7c __GI___pthread_kill
	./nptl/pthread_kill.c:89
0x7f5b03242475 __GI_raise
	../sysdeps/posix/raise.c:26
0x7f5b032287f2 __GI_abort
	./stdlib/abort.c:79
0x7f5b032896f5 __libc_message
	../sysdeps/posix/libc_fatal.c:155
0x7f5b032a0d7b malloc_printerr
	./malloc/malloc.c:5664
0x7f5b032a416b _int_malloc
	./malloc/malloc.c:4005
0x7f5b032a52e1 __GI___libc_malloc
	./malloc/malloc.c:3321
0x5627208cff70 allocate
	ccan/ccan/tal/tal.c:250
0x5627208d0619 tal_alloc_
	ccan/ccan/tal/tal.c:428
0x5627208d06ec tal_alloc_arr_
	ccan/ccan/tal/tal.c:471
0x562720814eb0 wally_tal
	common/setup.c:14
0x56272085419c wally_calloc
	../../../libwally-core/src/internal.c:304
0x5627208656b4 wally_tx_init_alloc
	../../../libwally-core/src/transaction.c:1137
0x56272086ad8a tx_from_bytes
	../../../libwally-core/src/transaction.c:2589
0x56272086b8d2 wally_tx_from_bytes
	../../../libwally-core/src/transaction.c:2744
0x562720821013 pull_wtx
	bitcoin/tx.c:573
0x562720822089 pull_bitcoin_tx
	bitcoin/tx.c:593
0x56272081af13 db_col_tx
	db/bindings.c:418
0x5627207f71f6 wallet_transactions_get
	wallet/wallet.c:4778
0x5627207f9573 json_listtransactions
	wallet/walletrpc.c:581
0x5627207bc8fa command_exec
	lightningd/jsonrpc.c:655
0x5627207bdf5d rpc_command_hook_final
	lightningd/jsonrpc.c:787
0x5627207e39b4 plugin_hook_call_
	lightningd/plugin_hook.c:288
0x5627207be4d5 plugin_hook_call_rpc_command
	lightningd/jsonrpc.c:875
0x5627207be4d5 parse_request
	lightningd/jsonrpc.c:988
0x5627207be89e read_json
	lightningd/jsonrpc.c:1093
0x5627208c77c9 next_plan
	ccan/ccan/io/io.c:59
0x5627208c7ca4 do_plan
	ccan/ccan/io/io.c:407
0x5627208c7d41 io_ready
	ccan/ccan/io/io.c:417
0x5627208c8f89 io_loop
	ccan/ccan/io/poll.c:453
0x5627207bc6b7 io_loop_with_timers
	lightningd/io_loop_with_timers.c:22
0x5627207c09e0 main
	lightningd/lightningd.c:1226
0x7f5b03229d8f __libc_start_call_main
	../sysdeps/nptl/libc_start_call_main.h:58
0x7f5b03229e3f __libc_start_main_impl
	../csu/libc-start.c:392
0x5627207a1c34 ???
	???:0
0xffffffffffffffff ???
	???:0
malloc(): invalid next size (unsorted)
Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Lost connection to the RPC socket.Aborted (core dumped)
```

**Comment by AutonomousOrganization** [comment]
Appears unrelated but got another crash. Could this mean hardware instability?
```
lightning_gossipd: gossipd/gossip_store.c:613: delete_by_index: Assertion `(be16_to_cpu(hdr.beflags) & GOSSIP_STORE_DELETED_BIT) == 0' failed.
lightning_gossipd: FATAL SIGNAL 6 (version v23.02.2)
0x55693ad21aa4 send_backtrace
	common/daemon.c:33
0x55693ad21b3b crashdump
	common/daemon.c:46
0x7f913ee4251f ???
	./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7f913ee96a7c __pthread_kill_implementation
	./nptl/pthread_kill.c:44
0x7f913ee96a7c __pthread_kill_internal
	./nptl/pthread_kill.c:78
0x7f913ee96a7c __GI___pthread_kill
	./nptl/pthread_kill.c:89
0x7f913ee42475 __GI_raise
	../sysdeps/posix/raise.c:26
0x7f913ee287f2 __GI_abort
	./stdlib/abort.c:79
0x7f913ee2871a __assert_fail_base
	./assert/assert.c:92
0x7f913ee39e95 __GI___assert_fail
	./assert/assert.c:101
0x55693ad14928 delete_by_index
	gossipd/gossip_store.c:613
0x55693ad155a9 gossip_store_delete
	gossipd/gossip_store.c:633
0x55693ad1d479 routing_expire_channels
	gossipd/routing.c:2250
0x55693ad12a52 new_blockheight
	gossipd/gossipd.c:820
0x55693ad13a24 recv_req
	gossipd/gossipd.c:1029
0x55693ad21bfe handle_read
	common/daemon_conn.c:35
0x55693adba895 next_plan
	ccan/ccan/io/io.c:59
0x55693adbad70 do_plan
	ccan/ccan/io/io.c:407
0x55693adbae0d io_ready
	ccan/ccan/io/io.c:417
0x55693adbc7aa io_loop
	ccan/ccan/io/poll.c:453
0x55693ad13f7a main
	gossipd/gossipd.c:1146
0x7f913ee29d8f __libc_start_call_main
	../sysdeps/nptl/libc_start_call_main.h:58
0x7f913ee29e3f __libc_start_main_impl
	../csu/libc-start.c:392
0x55693ad0f774 ???
	???:0
0xffffffffffffffff ???
	???:0
lightning_gossipd: FATAL SIGNAL (version v23.02.2)
0x55693ad21aa4 send_backtrace
	common/daemon.c:33
0x55693ad2852c status_failed
	common/status.c:221
0x55693ad2863e status_backtrace_exit
	common/subdaemon.c:18
0x55693ad21b41 crashdump
	common/daemon.c:49
0x7f913ee4251f ???
	./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0
0x7f913ee96a7c __pthread_kill_implementation
	./nptl/pthread_kill.c:44
0x7f913ee96a7c __pthread_kill_internal
	./nptl/pthread_kill.c:78
0x7f913ee96a7c __GI___pthread_kill
	./nptl/pthread_kill.c:89
0x7f913ee42475 __GI_raise
	../sysdeps/posix/raise.c:26
0x7f913ee287f2 __GI_abort
	./stdlib/abort.c:79
0x7f913ee2871a __assert_fail_base
	./assert/assert.c:92
0x7f913ee39e95 __GI___assert_fail
	./assert/assert.c:101
0x55693ad14928 delete_by_index
	gossipd/gossip_store.c:613
0x55693ad155a9 gossip_store_delete
	gossipd/gossip_store.c:633
0x55693ad1d479 routing_expire_channels
	gossipd/routing.c:2250
0x55693ad12a52 new_blockheight
	gossipd/gossipd.c:820
0x55693ad13a24 recv_req
	gossipd/gossipd.c:1029
0x55693ad21bfe handle_read
	common/daemon_conn.c:35
0x55693adba895 next_plan
	ccan/ccan/io/io.c:59
0x55693adbad70 do_plan
	ccan/ccan/io/io.c:407
0x55693adbae0d io_ready
	ccan/ccan/io/io.c:417
0x55693adbc7aa io_loop
	ccan/ccan/io/poll.c:453
0x55693ad13f7a main
	gossipd/gossipd.c:1146
0x7f913ee29d8f __libc_start_call_main
	../sysdeps/nptl/libc_start_call_main.h:58
0x7f913ee29e3f __libc_start_main_impl
	../csu/libc-start.c:392
0x55693ad0f774 ???
	???:0
0xffffffffffffffff ???
	???:0
2023-04-18T23:12:42.058Z **BROKEN** gossipd: FATAL SIGNAL 6 (version v23.02.2)
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: common/daemon.c:38 (send_backtrace) 0x55693ad21aec
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: common/daemon.c:46 (crashdump) 0x55693ad21b3b
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0 ((null)) 0x7f913ee4251f
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:44 (__pthread_kill_implementation) 0x7f913ee96a7c
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:78 (__pthread_kill_internal) 0x7f913ee96a7c
2023-04-18T23:12:42.058Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:89 (__GI___pthread_kill) 0x7f913ee96a7c
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ../sysdeps/posix/raise.c:26 (__GI_raise) 0x7f913ee42475
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ./stdlib/abort.c:79 (__GI_abort) 0x7f913ee287f2
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ./assert/assert.c:92 (__assert_fail_base) 0x7f913ee2871a
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ./assert/assert.c:101 (__GI___assert_fail) 0x7f913ee39e95
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/gossip_store.c:613 (delete_by_index) 0x55693ad14928
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/gossip_store.c:633 (gossip_store_delete) 0x55693ad155a9
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/routing.c:2250 (routing_expire_channels) 0x55693ad1d479
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:820 (new_blockheight) 0x55693ad12a52
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:1029 (recv_req) 0x55693ad13a24
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: common/daemon_conn.c:35 (handle_read) 0x55693ad21bfe
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x55693adba895
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x55693adbad70
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x55693adbae0d
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x55693adbc7aa
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:1146 (main) 0x55693ad13f7a
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ../sysdeps/nptl/libc_start_call_main.h:58 (__libc_start_call_main) 0x7f913ee29d8f
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ../csu/libc-start.c:392 (__libc_start_main_impl) 0x7f913ee29e3f
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: (null):0 ((null)) 0x55693ad0f774
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: (null):0 ((null)) 0xffffffffffffffff
2023-04-18T23:12:42.059Z **BROKEN** gossipd: FATAL SIGNAL (version v23.02.2)
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: common/daemon.c:38 (send_backtrace) 0x55693ad21aec
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: common/status.c:221 (status_failed) 0x55693ad2852c
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: common/subdaemon.c:18 (status_backtrace_exit) 0x55693ad2863e
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: common/daemon.c:49 (crashdump) 0x55693ad21b41
2023-04-18T23:12:42.059Z **BROKEN** gossipd: backtrace: ./signal/../sysdeps/unix/sysv/linux/x86_64/libc_sigaction.c:0 ((null)) 0x7f913ee4251f
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:44 (__pthread_kill_implementation) 0x7f913ee96a7c
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:78 (__pthread_kill_internal) 0x7f913ee96a7c
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./nptl/pthread_kill.c:89 (__GI___pthread_kill) 0x7f913ee96a7c
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ../sysdeps/posix/raise.c:26 (__GI_raise) 0x7f913ee42475
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./stdlib/abort.c:79 (__GI_abort) 0x7f913ee287f2
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./assert/assert.c:92 (__assert_fail_base) 0x7f913ee2871a
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ./assert/assert.c:101 (__GI___assert_fail) 0x7f913ee39e95
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/gossip_store.c:613 (delete_by_index) 0x55693ad14928
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/gossip_store.c:633 (gossip_store_delete) 0x55693ad155a9
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/routing.c:2250 (routing_expire_channels) 0x55693ad1d479
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:820 (new_blockheight) 0x55693ad12a52
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:1029 (recv_req) 0x55693ad13a24
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: common/daemon_conn.c:35 (handle_read) 0x55693ad21bfe
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:59 (next_plan) 0x55693adba895
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:407 (do_plan) 0x55693adbad70
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ccan/ccan/io/io.c:417 (io_ready) 0x55693adbae0d
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ccan/ccan/io/poll.c:453 (io_loop) 0x55693adbc7aa
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: gossipd/gossipd.c:1146 (main) 0x55693ad13f7a
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ../sysdeps/nptl/libc_start_call_main.h:58 (__libc_start_call_main) 0x7f913ee29d8f
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: ../csu/libc-start.c:392 (__libc_start_main_impl) 0x7f913ee29e3f
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: (null):0 ((null)) 0x55693ad0f774
2023-04-18T23:12:42.060Z **BROKEN** gossipd: backtrace: (null):0 ((null)) 0xffffffffffffffff
2023-04-18T23:12:42.060Z **BROKEN** gossipd: STATUS_FAIL_INTERNAL_ERROR: FATAL SIGNAL
lightningd: gossipd failed (exit status 242), exiting.

```

**Comment by AutonomousOrganization** [comment]
None >78500 :
```
[
  {
    "height": 786156,
    "hash": "00000000000000000003e76f5d8d166a1039af210afc4644dd15d4a61951a029",
    "branchlen": 0,
    "status": "active"
  },
  {
    "height": 784121,
    "hash": "000000000000000000046a2698233ed93bb5e74ba7d2146a68ddb0c2504c980d",
    "branchlen": 1,
    "status": "invalid"
  },
  {
    "height": 783830,
    "hash": "0000000000000000000366d2c12772a350f507879a5325203424e58ec440249b",
    "branchlen": 1,
    "status": "valid-fork"
  },
  {
    "height": 783478,
    "hash": "0000000000000000000446f7d3093688ae697386fed3f52a63812678ea6b251d",
    "branchlen": 1,
    "status": "valid-headers"
  },
  {
    "height": 771745,
    "hash": "00000000000000000004370a77a30add64bba97c26a90ca9643b45a75219b2a6",
    "branchlen": 1,
    "status": "valid-headers"
  },
  {
    "height": 771627,
    "hash": "00000000000000000005a92bdb6f9d55ea8d2b42579e0db6ca7764f97b6910e1",
    "branchlen": 1,
    "status": "valid-headers"
  }
]
```

**Comment by AutonomousOrganization** [comment]
fixed by v23.08
- Thank you !
